app/models/course/discussion/post.rb

```diff
@@ -1,5 +1,11 @@
 class Course::Discussion::Post < ActiveRecord::Base
   acts_as_forest order: :created_at
 
+  after_initialize :set_topic, if: :new_record?
```

Comment:
You are going to need to validate that if a parent is present, the topic of the parent and the topic of this post needs to be consistent.